flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/datagen/SequenceGenerator.java

```diff
@@ -83,9 +86,11 @@ public void open(
             final int taskIdx = runtimeContext.getIndexOfThisSubtask();
             final long congruence = start + taskIdx;
 
+            Preconditions.checkArgument(
```

Comment:
The validation check should happen as early as possible (in the location where these parameters are set, i.e. the constructor).

Reply:
Done.